src/overlays/actors/ovl_Demo_Geff/z_demo_geff.c

```diff
@@ -22,33 +46,180 @@ const ActorInit Demo_Geff_InitVars =
     (ActorFunc)DemoGeff_Update,
     (ActorFunc)DemoGeff_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/DemoGeff_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/DemoGeff_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80977EA8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80977F80.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80978030.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_809780E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_809781FC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_809782A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80978308.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80978344.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_80978370.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_809783D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/DemoGeff_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/func_809784D4.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Geff/DemoGeff_Draw.s")
+extern UNK_TYPE D_06000EA0;
+
+static void DemoGeff_Destroy(DemoGeff* this, GlobalContext* globalCtx) {
+
+}
+
+static void DemoGeff_Init(DemoGeff* this, GlobalContext* globalCtx) {
+    if (this->actor.params < 0 || this->actor.params >= 9) {
+        osSyncPrintf(VT_FGCOL(RED) "Demo_Geff_Actor_ct:arg_dataがおかしい!!!!!!!!!!!!\n" VT_RST);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->action = 0;
+    this->drawConfig = 0;
+}
+
+static void func_80977EA8(GlobalContext* globalCtx, u32 dlist) {
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    s16 pad;
+    Gfx* gfxArr[4];
```

Comment:
This applies to other functions but I would remove the pad and add an entry to gfxArr instead.
```suggestion
    Gfx* gfxArr[5];
```